db/warehouse/migrate/20250116145506_create_notification_configuration.rb

```diff
@@ -0,0 +1,24 @@
+class CreateNotificationConfiguration < ActiveRecord::Migration[7.0]
+  def change
+    create_table :notification_configurations do |t|
+      t.references :user, null: false
+      t.references :source, polymorphic: true, null: false
+      t.string :notification_slug, null: false, description: 'Class name for notification logic'
```

Comment:
the term 'slug' kind of trips me up here... usually a slug uniquely identifies a unique resource in a url. But this seems to be more of an event type. Also the class names don't appear to be usable. 
```suggestion
      t.string :notification_event, null: false, description: 'count_threshold_exceeded' | 'error_threshold_exceeded'
```

Reply:
Currently `notification_slug` is either `NotificationTypes::ImportRecordCountChangeThreshold` or `NotificationTypes::ImportErrorCountThreshold`.  Maybe I'm missing something.  We could change it to `notification_event`, I'm not totally opposed to that, let me see how invasive that would be.